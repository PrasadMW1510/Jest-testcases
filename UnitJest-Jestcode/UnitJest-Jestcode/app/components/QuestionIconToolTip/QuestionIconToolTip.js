/**
 *
 * QuestionIconToolTip
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import ReactTooltip from 'react-tooltip';

import './QuestionIconToolTip.scss';

function QuestionIconToolTip(props) {
  return (
    <span>
      <div className="question-icon" data-tip="" data-for="questionToolTip">
        &#63;
      </div>
      <ReactTooltip className="question-tool-tip" id="questionToolTip" effect="solid">
        {props.children}
      </ReactTooltip>
    </span>
  );
}

QuestionIconToolTip.propTypes = {
  children: PropTypes.node,
};

export default QuestionIconToolTip;
