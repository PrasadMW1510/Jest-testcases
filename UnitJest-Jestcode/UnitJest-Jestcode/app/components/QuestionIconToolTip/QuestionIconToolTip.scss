.question-icon {
  background-color: #efefef;
  border: solid 1px #dfded6;
  color: #4281ce;
  font-size: 10px;
  width: 18px;
  height: 18px;
  border-radius: 50%;
  text-align: center;
  display: inline-block;
  margin-left: 5px;
  padding-right: 1px;
  padding-top: 1px;
}

.question-tool-tip {
  /*
    we don't want to use !important
    but according to the tool tip docs
    this is how you override the default styles
   */
  font-size: 10px !important;
  background-color: #fffe9f !important;
  color: #000 !important;
  opacity: 1 !important;
  width: 345px;
  height: auto;

  &.place-top::after {
    border-top-color: #fffe9f !important;
    opacity: 1 !important;
  }
}
