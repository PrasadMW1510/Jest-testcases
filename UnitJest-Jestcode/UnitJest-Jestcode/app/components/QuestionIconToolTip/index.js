/**
 *
 * QuestionIconToolTip
 *
 */

export { default } from './QuestionIconToolTip';
