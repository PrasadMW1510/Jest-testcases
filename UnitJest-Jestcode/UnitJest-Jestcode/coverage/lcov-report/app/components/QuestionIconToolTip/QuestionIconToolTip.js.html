<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/QuestionIconToolTip/QuestionIconToolTip.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">app/components/QuestionIconToolTip</a> QuestionIconToolTip.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/8</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/**<span class="cstat-no" title="statement not covered" ></span>
 *
 * QuestionIconToolTip
 *
 */
&nbsp;
import React <span class="cstat-no" title="statement not covered" >from 'react';<span class="cstat-no" title="statement not covered" ></span></span>
import PropTypes <span class="cstat-no" title="statement not covered" >from 'prop-types';<span class="cstat-no" title="statement not covered" ></span></span>
import ReactTooltip <span class="cstat-no" title="statement not covered" >from 'react-tooltip';<span class="cstat-no" title="statement not covered" ></span></span>
&nbsp;
<span class="cstat-no" title="statement not covered" >import './QuestionIconToolTip.scss';<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span>
&nbsp;
function QuestionIconToolTip(props) {
  return (
    &lt;span&gt;<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span>
      &lt;div className="question-icon" data-tip="" data-for="questionToolTip"&gt;
        &amp;#63;
      &lt;/div&gt;
      &lt;ReactTooltip className="question-tool-tip" id="questionToolTip" effect="solid"&gt;
        {props.children}
      &lt;/ReactTooltip&gt;
    &lt;/span&gt;
  );
}
&nbsp;
<span class="cstat-no" title="statement not covered" >QuestionIconToolTip.propTypes = {</span>
  children: PropTypes.node,<span class="cstat-no" title="statement not covered" ></span>
};
&nbsp;
export default QuestionIconToolTip;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jul 21 2018 17:06:47 GMT+0530 (India Standard Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
