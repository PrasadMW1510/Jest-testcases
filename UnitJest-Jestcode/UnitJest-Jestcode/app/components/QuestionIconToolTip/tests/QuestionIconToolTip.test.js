import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import QuestionIconToolTip from '../index';

describe('<QuestionIconToolTip />', () => {
  let wrapper = null;

  beforeEach(() => {
    wrapper = shallow(<QuestionIconToolTip>Tool Tip Text</QuestionIconToolTip>);
  });

  it('expect the default to render', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
