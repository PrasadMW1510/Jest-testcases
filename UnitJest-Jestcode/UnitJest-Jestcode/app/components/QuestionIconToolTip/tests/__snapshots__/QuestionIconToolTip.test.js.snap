// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionIconToolTip /> expect the default to render 1`] = `
<span>
  <div
    className="question-icon"
    data-for="questionToolTip"
    data-tip=""
  >
    ?
  </div>
  <ReactTooltip
    className="question-tool-tip"
    effect="solid"
    id="questionToolTip"
    insecure={true}
    resizeHide={true}
    wrapper="div"
  >
    Tool Tip Text
  </ReactTooltip>
</span>
`;
